Updates for lightning receive

thisconnect opened the pull request:
Note to myself:
- copy button is missing

thisconnect committed:
frontend: use view for each recevie step

Change to use a View for each step separatelly so that we can use
ViewButtons container or also ViewHeader in the future.

thisconnect committed:
frontend: rename receive amount label to sats

thisconnect committed:
frontend: add invoice details and edit button to receive view

thisconnect committed:
frontend: update buttons and add summary on lightning receive

thisconnect committed:
frontend: fix center qrcode on desktop

thisconnect committed:
frontend: do not reset amount field in lightning receive

When going back from any step

thisconnect committed:
frontend: add invoice title on receive invoice step

thisconnect committed:
frontend: move optional label to its own component

thisconnect committed:
frontend: changed to fullscreen spinner and removed busy state

dangeross commented on frontends/web/src/routes/account/send/send.tsx:
```diff
@@ -652,9 +652,7 @@ class Send extends Component<Props, State> {
-                        <span className={style.labelDescription}>
-                          {t('note.input.description')}
-                        </span>
+                        <OptionalLabel>{t('note.input.description')}</OptionalLabel>
```
Should we change this to `lightning.send.description.optional`, or move this to `lightning.description.optional` ?

thisconnect replied:
This is not the lightning send but the onchain send component, that I refactored so we can use the same component+styles here and in lightning receive.

dangeross replied:
Hah, ok sorry. Didn't check the filename

dangeross commented on frontends/web/src/routes/lightning/receive/receive.tsx:
```diff
@@ -121,87 +122,116 @@ export function Receive({ accounts, code }: Props) {
-            </ColumnButtons>
-          </Column>
-        </Grid>
+        <Spinner text="Hardcode invoice creating going…" guideExists={false} />
```
Is this text shown? If so we should add a translation key

thisconnect replied:
Yes definitely, I changed to "Creating invoice"  and fixed it into the last commit.

dangeross replied:
Sounds good

dangeross commented on frontends/web/src/routes/lightning/receive/receive.tsx:
```diff
@@ -121,87 +122,116 @@ export function Receive({ accounts, code }: Props) {
+                  <QRCode data={receivePaymentResponse?.lnInvoice.bolt11} />
+                </div>
+                <div className={styles.invoiceSummary}>
+                  {amountSatsText} sats
+                  (--- EUR)
```
I imagine this is placeholder for the fiat conversion. FYI I have been tasked to include the conversion rates in the views also later.

thisconnect replied:
lets do fiat conversion after PR to upstream/staging-ln

thisconnect commented:
One more comment, in send PR I simplified some of the useEffect's and called parseInput directly. I think here we could also simplified it a bit. I didn't want to break anything but will look into reducing some of the state/effects soonish. (But just FYI, this should not block anything)

thisconnect commented:
@dangeross I think I addressed all of your comments PTAL

dangeross reviewed (approved):
LGTM
